### ISSUE TITLE
Compliance for cloud flows and for service principals

### ISSUE THREAD

**Issue by PPCan** [microsoft/coe-starter-kit#8669]
### Does this question already exist in our backlog?

- [X] I have checked and confirm this is a new question.

### What is your question?

I'm building a new developer centre in a Developer Environment, for the CoE Toolkit as our Makers only have E5 and can't use the included Dev Centre as it needs Premium.

I've installed the Toolkit in the Developer environment, and limited the compliance to just one environment that has a Canvas app for testing. It has been shared to 20 people, and the threshold has been set to 10.  This should trigger a compliance email.

It doesn't.

The issue seems to be that while there is data in the audit table there is nothing for this app, and therefore the flow doesn't think there is any issue.  Only LaunchApp events are present.

What's missing from the tables for the compliance flow to send the email..?

![image](https://github.com/user-attachments/assets/2214e9ca-f310-4b21-b55b-e734e3ad07d2)

![image](https://github.com/user-attachments/assets/6cc49e1a-647c-499a-9911-e6786f802214)

### What solution are you experiencing the issue with?

None

### What method are you using to get inventory and telemetry?

None

AB#3835

**Comment by Jenefer-Monroe** [comment]
In the CoE, sharing is not tracked by the audit logs, its gathered instead by this flow which only runs once every two weeks as its long running, so you'll need to ensure this has run before you can continue your testing: CLEANUP - Admin | Sync Template v3 (App Shared With)

**Comment by PPCan** [comment]
Ah! Ok.  Should for Testing I have a simple Canvas app, that is shared and I have launched.  I've sent the thresholds for compliance to 1 or less (except sharing) where possible.

Is that the only flow I should trigger to work through the different compliance scenario's?  

My testing steps are:
1- run driver
2 - run audit syncs
3 - run CLEAN UPs (All???)
4- what am I missing???

**Comment by PPCan** [comment]
I changed my mind on the sharing. :) 1 it is!
![image](https://github.com/user-attachments/assets/c0042897-261a-4bd4-98a2-915481d6d151)

**Comment by Jenefer-Monroe** [comment]
Seems like you have this resolved then yes? Please let me know if you need anything else!

**Comment by PPCan** [comment]
I could use some help understanding what flags some of the compliance 'tabs'.  For example, there are Model Driven apps listed if you choose 'All', but 'Missing Details' has no Model Driven apps listed.  This is the case for the other tabs too.

Is missing details is the only compliance item for Model Driven apps, why are there no Model Driven Apps listed?  And there is 

Also, all the compliance emails I got were for 'sharing', and nothing for launching, or groups etc.

Is there a list of what Power Platform 'type' (i.e Canvas, Model, Bot, Flow etc) and what compliance will be triggered?
![image](https://github.com/user-attachments/assets/f0a61381-0c11-4704-9161-ae4e68011b1d)


![image](https://github.com/user-attachments/assets/df2a858e-cc4d-44c0-b6d3-dc44f4bac380)

**Comment by PPCan** [comment]
Maybe a better way to put it is, which Flows/Tables Map to these environment variables considering they represent the compliance 'thresholds'.  I'm only getting emails for 'shared/published' compliance.
![image](https://github.com/user-attachments/assets/f71223ab-a6e0-4a92-ae91-0b8773f29605)

**Comment by PPCan** [comment]
Ok... after much digging.. it looks like the only items that are flagged for non compliance on a schedule are Canvas Apps and Chatbots.  Flows etc do not have a scheduled flow that checks for compliance and the compliance request has to be started from the Power Platform Admin View.

Is that correct?

**Comment by Jenefer-Monroe** [comment]
Hello that is correct yes. We just released flow shared with in July and it was not integrated with the DCC feature. Only chatbots and canvas apps have any automated flagging.

**Comment by PPCan** [comment]
Hmmm.  Okay, how do we see all non compliant items, so we can reach out to the owners from the Admin View?  Right now you have to select the specific record.  That's not practical.

**Comment by PPCan** [comment]
Also, we have a large Pro development side, which uses ALM with Service Principals and Service Accounts.  Email notifications are not going to work in that scenario, as Service Accounts are not typically Mailbox enabled and noone is monitoring them if they are.

What's a good approach to get an overview of all non compliance / inactive/ orphaned for Pro ALM implementations?

It feels like there is a whole module missing that can do Pro and cit dev management.

**Comment by Jenefer-Monroe** [comment]
Yes I think you're right. Unfortunately today we don't have a solution here. I can place this on the backlog to consider and hopefully resolve in the future. 
In the near term we are completely booked prepping for the next version of Data Export for more performant inventory, so we wont be able to work on this in the next few months.

**Comment by PPCan** [comment]
Ok great!

We can assess building something in the mean time, but what would you recommend in general for an approach?

At a minimum we need something that we can see all the apps/flows/bots that are non compliance/inactive/orphaned from one management module, just for that purpose.  The necessary fields are only available by app/flow/bot, as columns.  We could sort by fields by module I guess, but that's pretty painful...

**Comment by Jenefer-Monroe** [comment]
Sorry I would need to take some time to get my head around it before I could offer a recommendation. 
Please feel free to put your thoughts / plans here, and I can read them as you go to provide feedback as can the general community

**Comment by PPCan** [comment]
Ok! First question.

We can use a Power Pages Site, with authenticated page requests, to enable E5 users to access the CoE Toolkit dataverse.

The site would basically be a duplicate of the Model Driven app.

On a side note, I highly suggest replacing (or including a minimal version) of any user facing UX that is a Model Driven App, with Power Pages.  At least then we can use the authenticated Power Pages licensing to 'grant' premium access without having to license everyone, considering it's an unknown user base.

Back to the question. 

The Flow has the Developer Compliance Centre URL as an environment variable, however all the parameters are hard coded in the flow.  I don't want to modify the Flow as we are not in a position to take on managing changes to the Toolkit.

When I change the Dev Comp Centre URL to a Power Pages URL I get the below, which makes sense based on the text merge that is happening in the flow.  

However, what in the URL is needed to look up the record the email is referring too? We may be able to simply ignore the other  params, and use the Power Pages to load the correct record for the user to enter Biz justification etc.

Finally, will this approach work?  I can't think why not, unless there is something in the URL that Power Pages will never work with.

![image](https://github.com/user-attachments/assets/2d67ade9-53f3-4b6e-a418-0ae69364b8a1)

**Comment by Jenefer-Monroe** [comment]
I'm afraid I dont know the answer to questions about Power Pages as I dont know their licensing model nor am I that familiar with their capabilities.

**Comment by PPCan** [comment]
Fair enough.  Is there someone on the team that could speak to Power Pages?

**Comment by Jenefer-Monroe** [comment]
Yes I imagine you will be making quite a few changes to the flow so you will likely want to just make a copy to be the one you use. 
Here is our article on how to [extend the kit](https://learn.microsoft.com/power-platform/guidance/coe/modify-components). It shows you that you should create a new solution and so on. 

Regarding that url, you are correct it pieces together the MDA, the custom page, and the target ogbject as shown here
![image](https://github.com/user-attachments/assets/722a0802-5922-483e-a978-8710fc07e12b)

I hope that helps!

**Comment by Ayesha-320** [comment]
Hi @PPCan,
According to https://learn.microsoft.com/en-us/power-platform/alm/pipelines, Power Platform Pipelines are now the recommended approach for Application Lifecycle Management (ALM). The ALM Accelerator is no longer actively maintained or supported. Therefore, this issue is being closed accordingly.  

Best Regards, 
CoE Starter Kit Team
